eskip/string.go

```diff
@@ -80,91 +79,50 @@ func argsString(args []interface{}) string {
 	return strings.Join(sargs, ", ")
 }
 
-func sortedKeys[V interface{}](m map[string]V) []string {
-	keys := make([]string, 0, len(m))
-	for k := range m {
-		keys = append(keys, k)
+func sortChunk(s []string, from int) {
+	if len(s)-from > 1 {
+		sort.Strings(s[from:])
 	}
-	sort.Strings(keys)
-	return keys
 }
 
-func sortedCopy(values []string) []string {
-	toSortValues := make([]string, len(values))
-	copy(toSortValues, values)
-	sort.Strings(toSortValues)
-	return toSortValues
-}
-
-func (r *Route) predicateString(sortPredicates bool) string {
+func (r *Route) predicateString() string {
 	var predicates []string
 
 	if r.Path != "" {
 		predicates = appendFmtEscape(predicates, `Path("%s")`, `"`, r.Path)
 	}
 
-	hostRegexps := r.HostRegexps
-
-	if sortPredicates {
-		hostRegexps = sortedCopy(r.HostRegexps)
-	}
-
-	for _, h := range hostRegexps {
+	from := len(predicates)
+	for _, h := range r.HostRegexps {
```

Comment:
Makes sense to sort only Header and HeaderRegexp as others should have stable order